### PR TITLE
[13.0] stock_measuring_device improvements

### PR DESCRIPTION
**Pull request by mmequignon**
See atomic commits for informations

### CODE REVIEW

**Review by sebalix — APPROVED** [review]
LG, https://github.com/OCA/stock-logistics-warehouse/pull/1141 could be closed I suppose?
Also runbot is failing, to check.

**Review by TDu — APPROVED** [review]

File: stock_measuring_device/static/src/js/measuring_wizard.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                this.call("bus_service", "startPolling");
+            }
+        },
+        measuring_wizard_bus_notification: function(notifications) {
+            var self = this;
+            _.each(notifications, function(notification) {
+                var channel = notification[0];
+                var message = notification[1];
+                if (channel === "notify_measuring_wizard_screen") {
+                    switch (message.action) {
```
Comment (simahawk) [inline stock_measuring_device/static/src/js/measuring_wizard.js]: this switch seems useless as you are checking only one condition which can be done w/ an IF

File: stock_measuring_device/static/src/js/measuring_wizard.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                        case "refresh":
+                            self.measuring_wizard_bus_action_refresh(message.params);
+                            break;
+                    }
+                }
+            });
+        },
+        measuring_wizard_bus_action_refresh: function(params) {
+            var selectedIds = this.getSelectedIds();
+            if (!selectedIds.length) {
```
Comment (simahawk) [inline stock_measuring_device/static/src/js/measuring_wizard.js]:
```suggestion
            if (!selectedIds.length || params.model != this.modelName) {
```

File: stock_measuring_device/static/src/js/measuring_wizard.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                return;
+            }
+            var currentId = selectedIds[0];
+            if (params.id === currentId && params.model === this.modelName) {
```
Comment (simahawk) [inline stock_measuring_device/static/src/js/measuring_wizard.js]:
```suggestion
            if (params.id === currentId) {
```

File: stock_measuring_device/wizard/measuring_wizard.py [hunk 4]
```diff
@@ -149,6 +152,9 @@ def action_save(self):
 
     def action_close(self):
         self.ensure_one()
+        for line in self.line_ids.filtered(lambda l: l.scan_requested):
```
Comment (simahawk) [inline stock_measuring_device/wizard/measuring_wizard.py]: do the check in the for loop and avoid looping twice w/ filtered

File: stock_measuring_device/wizard/measuring_wizard.py [hunk 5]
```diff
@@ -158,11 +164,34 @@ def action_close(self):
...
+            self.product_id = pack.product_id
+            self.onchange_product_id()
+
     def reload(self):
         return {
             "type": "ir.actions.act_view_reload",
         }
 
+    def _send_notification_refresh(self):
+        """Send a refresh notification on the wizard.
```
Comment (simahawk) [inline stock_measuring_device/wizard/measuring_wizard.py:183]:
```suggestion
        """Send a refresh notification on the wizard.
        
```

**Review by simahawk — CHANGES_REQUESTED** [review]
squash and reword commits to include module name (eg: `measuring_device:`)

**Comment by mmequignon** [comment]
> squash and reword commits to include module name (eg: `measuring_device:`)

done :)

**Pull request merged**